### PR TITLE
#67: Port tests to JUnit5

### LINKED ISSUE #67: Port tests to JUnit5
# Situation
JUnit 5 is the successor of the JUnit 4 framework we are using at the moment. It has a better design, clearer life cycle handling and a uniform extension mechanism. In contrast JUnit 4 has two: Runners and Rules and they are not fully compatible.

# Acceptance Criteria
* Test suite is ported to JUnit 5
* Rules are replaced by extensions (six occurrences at the time of this writing)
* Runners are replaced by extensions (zero occurrences at the moment)
* POMs are updated
* Documentation is updated

### PR DESCRIPTION
**Pull request by redcatbear**
Closes #67.

Task-Url: https://github.com/exasol/virtual-schemas/issues/67

### CODE REVIEW

File: jdbc-adapter/virtualschema-jdbc-adapter/src/test/java/com/exasol/adapter/dialects/FileBasedIntegrationTest.java [hunk 1]
```diff
@@ -4,76 +4,78 @@
...
-        this.testFile = testFile;
+        return Arrays.stream(testDir.listFiles((dir, name) -> name.endsWith(".json")));
     }
 
-    @Test
-    public void testPushdownFromTestFile() throws Exception {
-        final String jsonTest = Files.toString(this.testFile, Charsets.UTF_8);
+    @MethodSource("listTestFiles")
+    @ParameterizedTest
+    void testPushdownFromTestFile(final File testFile) throws Exception {
```
Comment (AnastasiiaSergienko) [inline jdbc-adapter/virtualschema-jdbc-adapter/src/test/java/com/exasol/adapter/dialects/FileBasedIntegrationTest.java]: I think it's better to check and replace Exception with specific exceptions which are thrown

File: jdbc-adapter/virtualschema-jdbc-adapter/src/test/java/com/exasol/adapter/dialects/FileBasedIntegrationTest.java [hunk 2]
```diff
@@ -4,76 +4,78 @@
...
         for (int testNr = 0; testNr < numberOfTests; testNr++) {
             final List<PushDownRequest> PushDownRequests = getPushDownRequestsFrom(jsonTest, testNr);
             final Map<String, List<String>> expectedPushdownQueries = getExpectedPushdownQueriesFrom(jsonTest, testNr);
             for (final String dialect : expectedPushdownQueries.keySet()) {
                 for (final PushDownRequest PushDownRequest : PushDownRequests) {
-                    final String pushdownQuery = generatePushdownQuery(dialect, PushDownRequest,
-                            this.testFile.getName(), testNr);
-                    assertExpectedPushdowns(expectedPushdownQueries.get(dialect), pushdownQuery,
-                            this.testFile.getName(), testNr, dialect);
+                    final String pushdownQuery = generatePushdownQuery(dialect, PushDownRequest, testFile.getName(),
```
Comment (AnastasiiaSergienko) [inline jdbc-adapter/virtualschema-jdbc-adapter/src/test/java/com/exasol/adapter/dialects/FileBasedIntegrationTest.java]: Rename pushdownQuery -> pushDownQuery

File: jdbc-adapter/virtualschema-jdbc-adapter/src/test/java/com/exasol/adapter/dialects/FileBasedIntegrationTest.java (hunk 1 above)
Comment (redcatbear) [reply jdbc-adapter/virtualschema-jdbc-adapter/src/test/java/com/exasol/adapter/dialects/FileBasedIntegrationTest.java]: Done.

File: jdbc-adapter/virtualschema-jdbc-adapter/src/test/java/com/exasol/adapter/dialects/FileBasedIntegrationTest.java (hunk 2 above)
Comment (redcatbear) [reply jdbc-adapter/virtualschema-jdbc-adapter/src/test/java/com/exasol/adapter/dialects/FileBasedIntegrationTest.java]: Done.

**Pull request merged**